libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/model/eh/Edge.kt

```diff
@@ -0,0 +1,157 @@
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Edge
+
+        if (id != other.id) return false
+        if (level != other.level) return false
+        if (probability != other.probability) return false
+        if (heading != other.heading) return false
+        if (length != other.length) return false
+        if (out != other.out) return false
+        if (parent != other.parent) return false
+        if (frc != other.frc) return false
+        if (speed != other.speed) return false
+        if (ramp != other.ramp) return false
+        if (motorway != other.motorway) return false
+        if (bridge != other.bridge) return false
+        if (tunnel != other.tunnel) return false
+        if (toll != other.toll) return false
+        if (names != other.names) return false
+        if (curvature != other.curvature) return false
+        if (geometry != other.geometry) return false
+        if (speedLimit != other.speedLimit) return false
+        if (laneCount != other.laneCount) return false
+        if (meanElevation != other.meanElevation) return false
+        if (countryCode != other.countryCode) return false
+        if (stateCode != other.stateCode) return false
+
+        return true
+    }
+
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun hashCode(): Int {
+        var result = id.hashCode()
+        result = 31 * result + level.hashCode()
+        result = 31 * result + probability.hashCode()
+        result = 31 * result + heading.hashCode()
+        result = 31 * result + length.hashCode()
+        result = 31 * result + out.hashCode()
+        result = 31 * result + parent.hashCode()
+        result = 31 * result + frc.hashCode()
+        result = 31 * result + speed.hashCode()
+        result = 31 * result + ramp.hashCode()
+        result = 31 * result + motorway.hashCode()
+        result = 31 * result + bridge.hashCode()
+        result = 31 * result + tunnel.hashCode()
+        result = 31 * result + toll.hashCode()
+        result = 31 * result + names.hashCode()
+        result = 31 * result + curvature.hashCode()
+        result = 31 * result + geometry.hashCode()
+        result = 31 * result + speedLimit.hashCode()
+        result = 31 * result + laneCount.hashCode()
+        result = 31 * result + meanElevation.hashCode()
+        result = 31 * result + countryCode.hashCode()
+        result = 31 * result + stateCode.hashCode()
+        return result
+    }
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "Edge(" +
+            "id=$id, " +
+            "level=$level, " +
+            "probability=$probability, " +
+            "heading=$heading, " +
+            "length=$length, " +
+            "out=$out, " +
+            "parent=${parent?.id}, " +
```

Comment:
Run into `StackOverflow`s when testing this. Added the `parent.id` to avoid them. That made think if we should include the `Edge` object as `parent` or if we could make it work with the `id` modifying `ElectronicHorizonEdge.mapToEdge` and `mpp` logic 🤔 

Any ideas are more than welcome @etl @evabishchevich @SiarheiFedartsou @mskurydin @kmadsen

Reply:
what's the reason for having a reference to the `parent` edge?

Reply:
> - Add `parent` to `Edge` which makes it possible to traverse back (might be useful for a user wanting to traceback to the starting point)

Reply:
since you're asking for ideas: delete the `toString()` if we're expecting recursive objects

Reply:
another idea is to cover it with a test, `parent?.id` would be fine by me.